apis/hive/v1/clusterprovision_types.go

```diff
@@ -44,6 +44,9 @@ type ClusterProvisionSpec struct {
 
 	// PrevInfraID is the infra ID of the previous failed provision attempt.
 	PrevInfraID *string `json:"prevInfraID,omitempty"`
+
+	// PrevProvisionName is the name of the previous failed provision attempt.
+	PrevProvisionName *string `json:"prevProvisionName,omitempty"`
```

Comment:
I had to add this field to clear secrets from previously failed provisions. I tried avoiding it as part of this PR but that meant writing some weird-looking code that didn't make any sense. Something like if infraID != nil -> remove secrets but if prevInfraID != nil -> do not remove secrets.